src/jit/hwintrinsiclistxarch.h

```diff
@@ -144,6 +144,9 @@ HARDWARE_INTRINSIC(SSE2_ConvertToVector128Int32,                     "ConvertToV
 HARDWARE_INTRINSIC(SSE2_ConvertToVector128Int32WithTruncation,       "ConvertToVector128Int32WithTruncation",            SSE2,       -1,           16,          1,            {INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_cvttps2dq, INS_cvttpd2dq},         HW_Category_SimpleSIMD,                        HW_Flag_BaseTypeFromArg)
 HARDWARE_INTRINSIC(SSE2_ConvertToVector128Single,                    "ConvertToVector128Single",                         SSE2,       -1,           16,          1,            {INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_cvtdq2ps,  INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_cvtpd2ps},          HW_Category_SimpleSIMD,                        HW_Flag_BaseTypeFromArg)
 HARDWARE_INTRINSIC(SSE2_Divide,                                      "Divide",                                           SSE2,       -1,           16,          2,            {INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_divpd},             HW_Category_SimpleSIMD,                        HW_Flag_NoFlag)
+HARDWARE_INTRINSIC(SSE2_LoadAlignedVector128,                        "LoadAlignedVector128",                             SSE2,       -1,           16,          1,            {INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_movdqa,    INS_invalid,   INS_movapd},            HW_Category_MemoryLoad,                        HW_Flag_NoFlag)
+HARDWARE_INTRINSIC(SSE2_LoadScalarVector128,                         "LoadScalarVector128",                              SSE2,       -1,           16,          1,            {INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_movd,      INS_movd,      INS_movq,      INS_movq,      INS_invalid,   INS_movsdsse2},         HW_Category_MemoryLoad,                        HW_Flag_NoFlag)
```

Comment:
@fiigii, the 8-bit and 16-bit overloads in particular don't have any backing instruction support (there is no `movb` or `movw`, or equivalent that I can find), was this intentional?

Reply:
I have no idea about ISA design. AFAIK, the performance of partial register access is tricky on Intel architectures, especially <32-bit.

Reply:
I was asking since you exposed these particular LoadScalar APIs. I was wondering if it was intentional (and you expect a movd+clear unnecessary bits) or if it was just an oversight at the time (and we should remove these 4 APIs in particular).

Reply:
> it was just an oversight at the time

Yes, I think that is an oversight. `LoadScalarVector128` should just work on `float/double`.

Looks like we should remove integer `LoadScalarVector128` and add `Vector128<long/ulong> LoadLow(Vector128<long/ulong> upper, double* address)`.

Reply:
I think we want LoadScalar for int/unit and long/ulong.

The movd and movq instructions will explicitly clear upper, rather than setting it from one of the source operands

Reply:
> I think we want LoadScalar for int/unit and long/ulong.

Yes, I just looked that there are a few `Scalar` APIs work on int/long.

> The movd and movq instructions will explicitly clear upper, rather than setting it from one of the source operands

That is okay, which makes consistent with C++ intrinsic semantics.

Reply:
> That is okay, which makes consistent with C++ intrinsic semantics.

Yes, I was just indicating why we couldn't use them for `LoadLower` or `LoadUpper`, since those explicitly take a lower and upper value for the "other" bits.

I'll submit a separate PR to remove the 4 overloads that are not valid.

Reply:
@fiigii, did you have any other feedback or does this look good to you?

Reply:
https://github.com/dotnet/coreclr/pull/16221